apps/voting/contracts/Voting.sol

```diff
@@ -582,71 +632,56 @@ contract Voting is IForwarder, AragonApp {
-     * @param _voter address of the voter
+     * @param _delegate Address of the delegate
+     * @param _voter Address of the voter
      */
     function _addDelegatedAddressFor(address _delegate, address _voter) internal {
-        uint256 delegatedVotersCount = delegatedVoters[_delegate].addresses.length;
+        address[] storage votersList = delegatedVoters[_delegate].addresses;
+        uint256 delegatedVotersCount = votersList.length;
         require(delegatedVotersCount <= UINT_96_MAX, ERROR_MAX_DELEGATED_VOTERS_REACHED);
 
-        delegatedVoters[_delegate].addresses.push(_voter);
+        votersList.push(_voter);
         delegates[_voter] = Delegate(_delegate, uint96(delegatedVotersCount));
-        emit SetDelegate(_voter, _delegate);
+        emit AssignDelegate(_voter, _delegate);
     }
 
     /**
      * @dev internal function to remove  the `_voter` from the list of delegated voters for the `_delegate`
-     * @param _delegate address of the delegate
-     * @param _voter address of the voter
+     * @param _delegate Address of the delegate
+     * @param _voter Address of the voter
      */
     function _removeDelegatedAddressFor(address _delegate, address _voter) internal {
-        uint256 delegatedVotersCount = delegatedVoters[_delegate].addresses.length;
-        require(delegatedVotersCount > 0, ERROR_DELEGATE_NOT_SET);
-
         uint96 voterIndex = delegates[_voter].voterIndex;
-        assert(delegatedVoters[_delegate].addresses[voterIndex] == _voter);
-        address lastVoter = delegatedVoters[_delegate].addresses[delegatedVotersCount - 1];
-        if (voterIndex < delegatedVotersCount - 1) {
-            delegatedVoters[_delegate].addresses[voterIndex] = lastVoter;
+        address[] storage votersList = delegatedVoters[_delegate].addresses;
+        assert(votersList[voterIndex] == _voter);
+
+        uint256 lastVoterIndex = votersList.length - 1;
+        if (voterIndex < lastVoterIndex) {
+            address lastVoter = votersList[lastVoterIndex];
+            votersList[voterIndex] = lastVoter;
             delegates[lastVoter].voterIndex = voterIndex;
         }
-        delegatedVoters[_delegate].addresses.length--;
-        emit ResetDelegate(_voter, _delegate);
+        votersList.length--;
+        delete delegates[_voter];
+        emit UnassignDelegate(_voter, _delegate);
     }
 
     /**
-     * @notice Return the sliced list of delegated voters for `_delegate` and their voting power at the `_blockNumber`
-     * @param _delegate the address of the delegate
-     * @param _offset the number of delegated voters from the start of the list to skip
-     * @param _limit the number of delegated voters to return
-     * @param _blockNumber the block number to get the voting power
-     * @return the array of delegated voters
-     * @return the array of voting power of delegated voters
+     * @dev Internal function to get the voting power of the `_voters` at the `_blockNumber`
```

Comment:
"voting power" -> "cumulative voting power"

Reply:
Addressed in [19f14bc](https://github.com/lidofinance/aragon-apps/pull/44/commits/19f14bcc176a3d8c305cdffa708e19569110e683).